src/main/java/org/stellar/sdk/Predicate.java

```diff
@@ -186,15 +187,15 @@ public AbsBefore(TimePoint timePoint) {
     }
 
     public AbsBefore(long epochSeconds) {
-      this(new TimePoint(new Uint64(epochSeconds)));
+      this(new TimePoint(new Uint64(new XdrUnsignedHyperInteger(epochSeconds))));
     }
 
     public long getTimestampSeconds() {
-      return timePoint.getTimePoint().getUint64();
+      return timePoint.getTimePoint().getUint64().getNumber().longValue();
```

Comment:
this won't work well in cases where the value is larger than signed long can hold, coercing the biginteger into signed long has potential to force it to just the low-order of 64 bits:
example of issue - https://www.geeksforgeeks.org/biginteger-longvalue-method-in-java/
spec - https://docs.oracle.com/javase/8/docs/api/java/math/BigInteger.html#longValue--

I think the best approach is to preserve the original value in the the BigInteger and return it instead on these methods, such as:
`public BigInteger getTimestampSeconds()`

Reply:
Fixed in 8d28f047413d5d67855dcd556ef543f276c2b5ae